### PR TITLE
Fix 'No such file or directory' error in sigh resign

### PR DESCRIPTION
**Pull request by mgrebenets**
Fix for #7971 

Unnecessary quotes around `${KEYCHAIN_FLAG}` were causing an issue
instead of passing to `codesign` as `--keychain $KEYCHAIN_PATH` it was passed as `"--keychain $KEYCHAIN_PATH"` causing an issue.

An unfortunate side-effect of fixing shellcheck issues.
I have revisited all other changes in that shellcheck update and ran resigning to make sure it works.

### CODE REVIEW

**Review by hjanuschka — APPROVED** [review]

**Comment by mgrebenets** [comment]
ha :) so now shellcheck comlains abuout that change asking to double quote it :)

will add disable pragma

File: sigh/lib/assets/resign.sh [hunk 1]
```diff
@@ -764,7 +764,9 @@ function resign {
         log "Resigning application using certificate: '$CERTIFICATE'"
         log "and entitlements from provisioning profile: $NEW_PROVISION"
         cp -- "$TEMP_DIR/newEntitlements" "$APP_PATH/archived-expanded-entitlements.xcent"
-        /usr/bin/codesign ${VERBOSE} "${KEYCHAIN_FLAG}" -f -s "$CERTIFICATE" --entitlements "$TEMP_DIR/newEntitlements" "$APP_PATH"
+        # Must not qote KEYCHAIN_FLAG because it needs to be unwrapped and passed to codesign with spaces
+        # shellcheck disable=SC2086
+        /usr/bin/codesign ${VERBOSE} ${KEYCHAIN_FLAG} -f -s "$CERTIFICATE" --entitlements "$TEMP_DIR/newEntitlements" "$APP_PATH"
```
Comment (mfurtak) [inline sigh/lib/assets/resign.sh]: Is `"$CERTIFICATE"` going to have the same problem?

**Comment by asfalcone** [comment]
Does https://github.com/fastlane/fastlane/pull/7975/files#diff-7b2948da13b0a146580ebc7c06533c87L554 need to change as well?

**Comment by mgrebenets** [comment]
@asfalcone You mean the change to `sigh/lib/sigh/resign.rb`?

That one puts ipa name to be the last. That's more safe for parsing arguments in shell.
I looks for all the `a` `--aaa` flags and if none works - then assumes that's the output file name.
So I've put it to be last.

**Comment by mgrebenets** [comment]
@asfalcone I'm running more tests on all kinds of edge cases for resigning. Will update you soon. To be confident about releasing 2.10.1 patch.

**Comment by mgrebenets** [comment]
@asfalcone Saw it! One more quoting missing. PR is on the way.
